tests_e2e/tests/ext_update/ext_update_failure.py

```diff
@@ -0,0 +1,104 @@
+#
+
+import uuid
+from typing import List, Dict, Any
+
+from assertpy import fail, assert_that
+
+from tests_e2e.tests.lib.agent_test import AgentVmTest
+from tests_e2e.tests.lib.logging import log
+from tests_e2e.tests.lib.ssh_client import SshClient
+from tests_e2e.tests.lib.virtual_machine_extension_client import VirtualMachineExtensionClient
+from tests_e2e.tests.lib.vm_extension_identifier import VmExtensionIds, VmExtensionIdentifier
+
+
+class ExtensionUpdateFailureTest(AgentVmTest):
+    """
+    This test verifies that the agent reports correct status and code when extension operation fails during the ext update.
+    """
+
+    def run(self):
+        ssh_client: SshClient = self._context.create_ssh_client()
+        is_arm64: bool = ssh_client.get_architecture() == "aarch64"
+
+        custom_script_2_0 = VirtualMachineExtensionClient(
+            self._context.vm,
+            VmExtensionIds.CustomScript,
+            resource_name="CustomScript")
+
+        if is_arm64:
+            log.info("Will skip the update scenario, since currently there is only 1 version of CSE on ARM64")
+            return
+        else:
+            log.info("Installing %s", custom_script_2_0)
+            message = f"Hello {uuid.uuid4()}!"
+            custom_script_2_0.enable(
+                protected_settings={
+                    'commandToExecute': f"echo \'{message}\'"
+                },
+                auto_upgrade_minor_version=False
+            )
+            custom_script_2_0.assert_instance_view(expected_version="2.0", expected_message=message)
+
+        log.info("Modifying existing handler commands (disable operation) in HandlerManifest.json to fail the disable operation during update")
+
+        output = ssh_client.run_command(f"ext_update-modify_handler_manifest.py --extension-name '{custom_script_2_0._identifier}' --cmd-name disableCommand --cmd-value 'disablefailed'", use_sudo=True)
+        log.info("Modified handlerManifest.json:\n%s", output)
+
+        custom_script_2_1 = VirtualMachineExtensionClient(
+            self._context.vm,
+            VmExtensionIdentifier(VmExtensionIds.CustomScript.publisher, VmExtensionIds.CustomScript.type, "2.1"),
+            resource_name="CustomScript")
+
+        log.info("Updating %s", custom_script_2_0)
+
+        message = f"Hello {uuid.uuid4()}!"
+        try:
+            custom_script_2_1.enable(
+                protected_settings={
+                    'commandToExecute': f"echo \'{message}\'"
+                }
+            )
+            fail("The agent should have reported an error processing the extension update")
+        except Exception as error:
+            assert_that(str(error)).described_as(f"Expected VMExtensionHandlerNonTransientError/ExtensionUpdateError for {custom_script_2_1._identifier}, but actual error was: {error}").contains("VMExtensionHandlerNonTransientError").contains(f"{custom_script_2_1._identifier}").contains("ExtensionUpdateError")
```

Comment:
Ext status shows as transitioning: 
<img width="639" height="923" alt="image" src="https://github.com/user-attachments/assets/2503c4dd-f07f-41af-b975-0dc27bb3ccd6" />
Shouldn't it be failed?

Reply:
Don't know how transitioning coming in the portal, we dont report that in status. As far CRP operation, its failed with error immediately

Reply:
did it show as extension not finished execution?

Reply:
I don't know who parses the report status, they may need to change it.

Reply:
It may be worth checking with CRP folks

Reply:
I checked with Joe on this; he pointed that in general handler status of NotReady maps to an extension status of Transitioning in instanceview. This is unrelated to update scenario. Whether its need an update or not is a separate discussion I guess.

https://msazure.visualstudio.com/One/_git/Compute-CPlat-Core?path=/src/CRP/Dev/ComputeResourceProvider/Service/API/Operations/InVMArtifactsInstanceViewHelpers.cs&version=GBmaster&_a=contents